### DISCUSSION TITLE
Commit All and Sync: Wrong Version Number Referenced

### DISCUSSION THREAD

**Discussion by CodegoErgoSum** [community/community#48837] [Actions] [answered]
### Select Topic Area

Question

### Body

Hi, Everybody!...

I messed up when I last **Commit All and Sync**.  The message referenced the WRONG release version number: [TMGServices v1.3.4 (LOCAL)](https://github.com/CodegoErgoSum/TMGServices/commit/9c6f554773128251c91df2bc6348e5e31bec3dc4).  It should have been as follows: **TMGServices v1.3._5_ (LOCAL)**.

How can I fix that?...  Please advise.

![image](https://user-images.githubusercontent.com/13341617/221967952-296c6bbe-775d-4aac-8925-13512a98924b.png)

**Comment by vishnubansal001** [comment, marked as answer]
Hi @CodegoErgoSum bro,
To fix the commit message for your previous commit, you can use the following steps:

1. Open your command prompt or terminal and navigate to the local repository where you made the incorrect commit.
2. Use the following command to open the commit message in your default text editor: `git commit --amend`
3. The text editor will open with the commit message that you want to modify. Change the message to the correct version number, in this case "TMGServices v1.3.5 (LOCAL)".
4. Save the modified commit message and close the text editor.
5. Use the following command to force push the modified commit to your remote repository:`git push --force origin` Note that the "--force" flag is used to overwrite the previous commit on the remote repository with the modified commit.
6. Verify that the commit message has been updated correctly by checking the commit history on the remote repository.

Remember, it's important to use caution when using the "--force" flag, as it can potentially overwrite changes made by other collaborators. It's best to communicate with your team members and ensure that they are aware of any changes you make to the repository.

**Reply by CodegoErgoSum** [reply]
Hi, Vishnu...

Thank-you for your prompt response.

Upon executing the second step, an unfamiliar text editor appears.  It displays the commit message in multi-colored text with the following lines...

_Note: I injected a single quote ("'") at the beginning of each line to avoid having each line interpreted as a **header text**._

> '# Please enter the commit message for your changes. Lines starting
> '# with '#' will be ignored, and an empty message aborts the commit.
> '#
> '# Date:      Fri Feb 17 13:54:06 2023 -0500
> '#
> '# On branch master
> '# Your branch is up-to-date with 'origin/master'.
> '#
> `<SNIP />`

The following two lines also appear at the VERY bottom of my command prompt window (I cannot navigate to them)...

> ~/source/repos/<reposName>/.git/COMMIT_EDITMSG [unix] (08:29 01/03/2023)
> "<reposPath>/.git/COMMIT_EDITMSG" [unix] 53L, 3076C

I have no clue as to where/how to "enter the commit message for your changes" or how to exit, despite the instructions above.  I typed several keystrokes, but no change was visible.  I tried to "cut 'n' paste" (`Control-C`/`Control-V`)...  Select `<Backspace>`...  Select `<Delete>`...  Finally, I repeatedly `Control-Z`, hoping to "undo" any "changes" I might have incurred.  Then I simply closed the command prompt window.

I hope that does not "corrupt" my commit message, nor leave it in some "limbo" state.

Is there a way to change my "default" editor to one that I am familiar?...  Or do I need to learn git's "native" editor commands?

**Comment by CodegoErgoSum** [comment]
Hi, Vishnu!...

I have GREAT news!

I found out how to change my "default" editor to NotePad++.  I then followed your instructions and was, therefore, successful in updating the commit message.

Thank-you, bruh!

Have a GREAT day!

**Comment by CodegoErgoSum** [comment]
This issue has been resolved.
